### DISCUSSION TITLE
Why, with SSR, does the "/" route have to be prerendered (SSG) by default?

### DISCUSSION THREAD

**Discussion by nunoarruda** [analogjs/analog#944] [Q&A] [answered]
According to the [Prerendering routes](https://analogjs.org/docs/features/server/server-side-rendering#prerendering-routes) documentation:

> With SSR, the "/" route is prerendered by default.
> 
> It is a necessary step to return a rendered HTML when the user visits the root of the application. The prerendered routes can be customized, but keep in mind to include the "/" route also.

Why is that?
Wouldn't that mean that data would only be updated on the next build?
What if my "/" route (homepage / default route) has dynamic data or content that changes frequently and I want to SSR it?

I know I could just disable SSR and SSG and use the app as a regular CSR/SPA app, but my goal here is to use SSR on the homepage.

Thanks in advance for any clarification 👍

**Comment by brandonroberts** [comment, marked as answer]
The `/` is prerendered by default because most of the sites built with Analog are more content focused, so the / route is likely a landing page. You don't have to disable SSR to have a dynamically rendered / page.

Just pass an empty array to the `routes` array and it will use SSR for initial page requests and switch to CSR/SPA mode for navigation after that so you get the best of both.

**Reply by osnoser1** [reply]
I was looking for the repo about this question because I'm rendering dynamic content on the homepage. I didn't know about the pre-rendered '/' until I changed the content, and it wasn't reflected on the page. I'm using this config `analog({ prerender: { routes: [] } }),` but for some reason, I deactivate the SSR in the '/' page, but it keeps working in other routes; do you know why it could be happening this? Is it expected?

**Reply by brandonroberts** [reply]
That's because SSR is still enabled with that configuration, its only prerendering of those routes that's disabled. You can set `ssr: false` and `static: true` to completely turn off SSR and just build an SPA. If you do that then you'll have to handle the fallback of redirecting every request back to `/index.html` as usual.

**Reply by osnoser1** [reply]
Oh, sorry, I believe I didn't explain myself well. I'm using Sanity as a CMS in an angular app with Analog.js and all the content of the pages, including the homepage, is dynamic, I need to apply SSR on them, but the problem is the '/' page is always prerendering, and when I try to turn off SSG with the config I mentioned before, all the pages use SSR  (as expected) except the '/' page (not expected), and that's is a problem because the content of the homepage is not updating it. 

In summary, I need a complete SSR app enabled on all the pages without SSG, but I couldn't turn off the SSG on the '/' page without turning off the SSR, which is weird.

**Reply by brandonroberts** [reply]
Oh ok. That's because the `dist/analog/public/index.html` is created when building the client-side app. If you want to use SSR for the `/` route, you'll need to disable it in the `analog` plugin itself. 

```ts
      analog({
        nitro: {
          routeRules: {
            '/': {
              prerender: false,
            },
          },
        },
      }),
```

This is also done in apps that use tRPC

Example: https://github.com/analogjs/analog/blob/beta/apps/trpc-app/vite.config.ts

**Reply by osnoser1** [reply]
Ohh, it worked @brandonroberts, thank you 😄
